network/nattype.go

```diff
@@ -0,0 +1,31 @@
+package network
+
+// NATDeviceType indicates the type of the NAT device.
+type NATDeviceType int
+
+const (
+	// NATDeviceTypeUnknown indicates that the type of the NAT device is unknown.
+	NATDeviceTypeUnknown NATDeviceType = iota
+
+	// NATDeviceTypeCone indicates that the NAT device is a Cone NAT.
+	// A Cone NAT is a NAT where all outgoing connections from the same source IP address and port are mapped by the NAT device
+	// to the same IP address and port irrespective of the destination address.
+	// With Regards to Internet Society RFC 3489, this could be either a Full Cone NAT, a Restricted Cone NAT or a
+	// Port Restricted Cone NAT. However, we do NOT differentiate between them here and simply classify all such NATs as a Cone NAT.
+	// NAT traversal with hole punching is possible with a Cone NAT if the remote peer is ALSO behind a Cone NAT.
```

Comment:
```suggestion
	// NAT traversal with hole punching is possible with a Cone NAT, even if the remote peer is also behind a Cone NAT.
```

Reply:
@marten-seemann 

I've updated the comment to reflect that hole punching will work ONLY if the remote peer is also behind a Cone NAT and fail if the remote peer is behind a Symmetric NAT.